package/yapsy/PluginManager.py

```diff
@@ -577,11 +574,17 @@ def _importModule(plugin_module_name, candidate_filepath):
 		.. note:: Isolated and provided to be reused, but not to be reimplemented !
 		"""
 		# use imp to correctly load the plugin as a module
+		candidate_module = None
 		if os.path.isdir(candidate_filepath):
-			candidate_module = imp.load_module(plugin_module_name,None,candidate_filepath,("py","r",imp.PKG_DIRECTORY))
+			if (spec := importlib.util.spec_from_file_location(candidate_filepath.split('/')[-1], candidate_filepath + "/__init__.py")) is not None:
+				candidate_module = importlib.util.module_from_spec(spec)
+				sys.modules[plugin_module_name] = candidate_module
+				spec.loader.exec_module(candidate_module)
 		else:
-			with open(candidate_filepath+".py","r") as plugin_file:
-				candidate_module = imp.load_module(plugin_module_name,plugin_file,candidate_filepath+".py",("py","r",imp.PY_SOURCE))
+			if (spec := importlib.util.spec_from_file_location(candidate_filepath.split('/')[-1], candidate_filepath + ".py")) is not None:
+				candidate_module = importlib.util.module_from_spec(spec)
+				sys.modules[plugin_module_name] = candidate_module
+				spec.loader.exec_module(candidate_module)
```

Comment:
Legibility suggestion: this would be easier to understand and offer a cleaner traceback (if/upon error) if refactored into separate path munging and conditional spec assignment/exec. Does this seem logically equivalent?
```python
candidate_module = None
filepath_base = candidate_filepath.split('/')[-1]
if os.path.isdir(candidate_filepath):
    location = candidate_filepath + '/__init__.py'
else:
    location = candidate_filepath + '.py'

if (spec := importlib.util.spec_from_file_location(filepath_base, location)):
    candidate_module = importlib.util.module_from_spec(spec)
    sys.modules[plugin_module_name] = candidate_module
    spec.loader.exec_module(candidate_module)

return candidate_module
```

Reply:
Makes sense, update the PR with the suggestions.